pennylane/qchem/molecule.py

```diff
@@ -130,6 +131,17 @@ def __init__(
 
         self.mo_coefficients = None
 
+    def __repr__(self):
+        """Returns the molecule representation in string format"""
+
+        elements, counter, flags = set(self.symbols), collections.Counter(self.symbols), []
+        if counter["C"]:
+            flags = ["C", "H"] if counter["H"] else ["C"]
+        ordered_elems = flags + list(sorted(elements.difference(set(flags))))
+        formula = "".join([x + str(counter[x]) if counter[x] > 1 else x for x in ordered_elems])
```

Comment:
I guess you are following the Hill system for naming the molecule which is a bit dangerous! For example, lithium hydride will be shown as HLi and ammonia as H3N. I am fine with this but another option will be just listing the symbols.

Reply:
Yep, I am following Hill's notation. The only reason not to use symbols directly is that it would make the representation way too long on any compound with more than two atoms.